metrics/config_observability.go

```diff
@@ -79,13 +79,22 @@ type ObservabilityConfig struct {
 	// MetricsCollectorAddress specifies the metrics collector address. This is only used
 	// when the metrics backend is opencensus.
 	MetricsCollectorAddress string
+
+	// EnableDeprecatedMetricPrefix retains the legacy behaviour of prefixing our metric names
+	// with either the component name and/or metric domain
+	//
+	// When enabled:
+	// - OpenCensus exporter prefixes the domain and component
+	// - Prometheus exporter prefixes the component name (as a prom namespace)
+	EnableDeprecatedMetricPrefix bool
```

Comment:
Since this is a breaking change and I see there is a plan [here](https://github.com/knative/pkg/issues/2174#issuecomment-875009789) for mirgation, next question would be if there is a plan for removing this flag or users should just keep this flag on.

Reply:
Probably worth surfacing this topic on the thread.